csharp/ql/src/Stubs/Stubs.qll

```diff
@@ -260,105 +230,164 @@ private class GeneratedNamespace extends Namespace, GeneratedElement {
 }
 
 private string stubQualifiedNamePrefix(ValueOrRefType t) {
-  if t.getParent() instanceof GlobalNamespace then result=""
-  else if t.getParent() instanceof Namespace then result = t.getParent().(Namespace).getQualifiedName() + "."
-  else result = stubQualifiedNamePrefix(t.getParent()) + "."
+  if t.getParent() instanceof GlobalNamespace
+  then result = ""
+  else
+    if t.getParent() instanceof Namespace
+    then result = t.getParent().(Namespace).getQualifiedName() + "."
+    else result = stubQualifiedNamePrefix(t.getParent()) + "."
 }
 
 language[monotonicAggregates]
 private string stubClassName(Type t) {
-  if t instanceof ObjectType then
-    result = "object"
-  else if t instanceof StringType then
-    result = "string"
-  else if t instanceof IntType then
-    result = "int"
-  else if t instanceof BoolType then
-    result = "bool"
-  else if t instanceof VoidType then
-    result = "void"
-  else if t instanceof FloatType then
-    result = "float"
-  else if t instanceof DoubleType then
-    result = "double"
-  else if t instanceof NullableType then
-    result = stubClassName(t.(NullableType).getUnderlyingType()) + "?"
-  else if t instanceof TypeParameter then
-    result = t.getName()
-  else if t instanceof ArrayType then
-    result = stubClassName(t.(ArrayType).getElementType()) + "[]"
-  else if t instanceof PointerType then
-    result = stubClassName(t.(PointerType).getReferentType()) + "*"
-  else if t instanceof TupleType then
-    result = "(" + concat(int i, Type element | element = t.(TupleType).getElementType(i) | stubClassName(element), "," order by i) + ")"
-  else if t instanceof ValueOrRefType then
-    result = stubQualifiedNamePrefix(t) +  t.getUndecoratedName() + stubGenericArguments(t)
+  if t instanceof ObjectType
+  then result = "object"
   else
-    result = "<error>"
+    if t instanceof StringType
+    then result = "string"
+    else
+      if t instanceof IntType
+      then result = "int"
+      else
+        if t instanceof BoolType
+        then result = "bool"
+        else
+          if t instanceof VoidType
+          then result = "void"
+          else
+            if t instanceof FloatType
+            then result = "float"
+            else
+              if t instanceof DoubleType
+              then result = "double"
+              else
+                if t instanceof NullableType
+                then result = stubClassName(t.(NullableType).getUnderlyingType()) + "?"
+                else
+                  if t instanceof TypeParameter
+                  then result = t.getName()
+                  else
+                    if t instanceof ArrayType
+                    then result = stubClassName(t.(ArrayType).getElementType()) + "[]"
+                    else
+                      if t instanceof PointerType
+                      then result = stubClassName(t.(PointerType).getReferentType()) + "*"
+                      else
```

Comment:
@aschackmull Are we happy with this formatting?

Reply:
Yes. It actually shows you very directly how deeply nested those final if-then-elses are - the generated DIL is going to be similarly nested.  If the cases are disjoint you could change it to a disjunction, which will generally perform better.